ads/springAds.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createCustomEvent} from '../src/event-helper';
+import {loadScript} from '../3p/3p';
+import {parseJson} from '../src/json';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function springAds(global, data) {
+  context.master.availableSlots = context.master.availableSlots || {};
```

Comment:
could you check if the following method meets your needs for master frame selection:

https://github.com/ampproject/amphtml/blob/a04bd32fda335cc87cc6e5be29580b837d924f4f/3p/3p.js#L172-L186

Reply:
Goal is to have the master frame as director/handler, that's why a reference to the other adframes is needed